pkg/testutils/logger_test.go

```diff
@@ -45,3 +46,40 @@ func TestJSONLineError(t *testing.T) {
 	_, ok := log.JSONLine(0)["error"]
 	assert.True(t, ok, "must have 'error' key")
 }
+
+var keepLogging = func(logger *zap.Logger, quit chan struct{}) {
+	for {
+		select {
+		case <-quit:
+			return
+		default:
+			logger.Info("string")
+		}
+	}
+}
+
+var keepReading = func(buffer *Buffer, quit chan struct{}) {
+	for {
+		select {
+		case <-quit:
+			return
+		default:
+			buffer.Lines()
+			buffer.Stripped()
+		}
+	}
+}
+
+// This test should complete without any panic
+func TestRaceCondition(t *testing.T) {
+	logger, buffer := NewLogger()
+	logQuit := make(chan struct{})
+	readQuit := make(chan struct{})
+	defer close(logQuit)
+	defer close(readQuit)
+
+	go keepLogging(logger, logQuit)
+	go keepReading(buffer, readQuit)
+
+	time.Sleep(time.Second)
```

Comment:
is this necessary? This is a test for a _test_ facility, and the test takes more time than tests for whole business packages.